pymatgen/io/lobster/inputs.py

```diff
@@ -551,10 +555,9 @@ def write_KPOINTS(
                 kpts.append(f)
                 weights.append(0.0)
                 all_labels.append(labels[k])
-        ISYM = isym
-        comment = f"{ISYM=}, grid: {mesh} plus kpoint path" if line_mode else f"{ISYM=}, grid: {mesh}"
+        comment = f"{isym=}, grid: {mesh} plus kpoint path" if line_mode else f"{isym=}, grid: {mesh}"
 
-        KpointObject = Kpoints(
+        kpoint_object = Kpoints(
```

Comment:
I renamed a few internal variable to snake_case @JaGeo, let me know if you prefer not to change them :)

Reply:
It's weird, two of the `io.lobster.inputs` tests failed on my local test but passed in CI:
```python
(venv) yang@Yang-MacBook pymatgen % pytest tests/io/lobster/test_inputs.py             
........................F........F...........................                                                              [100%]
============================================================ FAILURES ============================================================
_____________________________________________ TestLobsterin.test_dict_functionality ______________________________________________

self = <lobster.test_inputs.TestLobsterin testMethod=test_dict_functionality>

    def test_dict_functionality(self):
>       assert self.Lobsterinfromfile.get("COHPstartEnergy") == -15.0
E       AssertionError

tests/io/lobster/test_inputs.py:1773: AssertionError
____________________________________________ TestLobsterin.test_read_write_lobsterin _____________________________________________

self = <lobster.test_inputs.TestLobsterin testMethod=test_read_write_lobsterin>

    def test_read_write_lobsterin(self):
        outfile_path = tempfile.mkstemp()[1]
        lobsterin1 = Lobsterin.from_file(f"{TEST_FILES_DIR}/cohp/lobsterin.1")
        lobsterin1.write_lobsterin(outfile_path)
>       lobsterin2 = Lobsterin.from_file(outfile_path)

tests/io/lobster/test_inputs.py:1793: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

cls = <class 'pymatgen.io.lobster.inputs.Lobsterin'>, lobsterin = '/var/folders/nn/rmcc0j095h53jgz4gj734q900000gn/T/tmpf82ed5jw'

    @classmethod
    def from_file(cls, lobsterin: str) -> Self:
        """
        Args:
            lobsterin (str): path to lobsterin.
    
        Returns:
            Lobsterin object
        """
        with zopen(lobsterin, mode="rt") as file:
            data = file.read().split("\n")
        if len(data) == 0:
            raise OSError("lobsterin file contains no data.")
        Lobsterindict: dict[str, Any] = {}
    
        for datum in data:
            # Remove all comments
            if not datum.startswith(("!", "#", "//")):
                pattern = r"\b[^!#//]+"  # exclude comments after commands
                if matched_pattern := re.findall(pattern, datum):
                    raw_datum = matched_pattern[0].replace("\t", " ")  # handle tab in between and end of command
                    key_word = raw_datum.strip().split(" ")  # extract keyword
                    if len(key_word) > 1:
                        # check which type of keyword this is, handle accordingly
                        if key_word[0].lower() not in [datum2.lower() for datum2 in Lobsterin.LISTKEYWORDS]:
                            if key_word[0].lower() not in [datum2.lower() for datum2 in Lobsterin.FLOAT_KEYWORDS]:
                                if key_word[0].lower() not in Lobsterindict:
                                    Lobsterindict[key_word[0].lower()] = " ".join(key_word[1:])
                                else:
                                    raise ValueError(f"Same keyword {key_word[0].lower()} twice!")
                            elif key_word[0].lower() not in Lobsterindict:
>                               Lobsterindict[key_word[0].lower()] = float(key_word[1])
E                               ValueError: could not convert string to float: 'None'

pymatgen/io/lobster/inputs.py:599: ValueError
```

Reply:
Not sure why they are failing now. Why is there suddenly a 'None' where there should be a float.

Reply:
I don't know... I have to assume there is something wrong with either my machine or there is some platform-related issue (as I just tried to run test on the unmodified master branch and still got the same error, so I assume this is not related to this PR). Is test failing on your local machine?

Also I tried to add explicit encoding to both `io.lobster.inputs` and its unit tests, but didn't fix the problem.

Reply:
Okay, I see. Yes, could be platform related. Will check again on my machine but maybe tomorrow. 😅 I also feel some of the code needs a bit of refactoring that I don't have time for at the moment.

Reply:
I don't have this error on Linux at all. Not sure how this happens. Could it be related to the files?

Reply:
I haven't got time to look closer into the reason sorry. I would ping you if I find anything. 

I just tried to run tests on CodeSpace (Linux runner) and no tests fail. I would assume there's something wrong with my machine but I don't know what exactly is. 

Actually I have met with similar issues where the `numpy` is missing and `ImportError` is suppressed, but unfortunately it isn't this time. I would have a look soon and see if I could find anything :)

Reply:
Thanks!
Maybe a change in the test files format?

Reply:
No problem at all!

> Maybe a change in the test files format?

I tried to explicitly set the encoding in both `lobster.inputs` and its test which unfortunately doesn't help.

I prefer not to make too many guesses before I actually look into it, to avoid causing confusion to you and everyone else who might be watching. 

For some reason I always notice unit test fails on my local machine (OS set to English, though I speak Mandarin but I work/study in English) but passed in CI, it's super weird as I thought MacOS is also Unix-based, but could also suggest some inconsistency. And that's the reason I always add explicit encoding as `utf-8` to `with open()`, which is also suggested by `pylint`. I had a discussion with Janosh on this before, as the default varies from platform to platform. From the [official docs](https://docs.python.org/3/library/functions.html#open):
> encoding is the name of the encoding used to decode or encode the file. This should only be used in text mode. The default encoding is platform dependent (whatever [locale.getencoding()](https://docs.python.org/3/library/locale.html#locale.getencoding) returns), but any [text encoding](https://docs.python.org/3/glossary.html#term-text-encoding) supported by Python can be used. See the [codecs](https://docs.python.org/3/library/codecs.html#module-codecs) module for the list of supported encodings.

and 

>locale.getencoding()
    Get the current [locale encoding](https://docs.python.org/3/glossary.html#term-locale-encoding):
        On Android and VxWorks, return "utf-8".
        On Unix, return the encoding of the current [LC_CTYPE](https://docs.python.org/3/library/locale.html#locale.LC_CTYPE) locale. Return "utf-8" if nl_langinfo(CODESET) returns an empty string: for example, if the current LC_CTYPE locale is not supported.
        On Windows, return the ANSI code page.

I'm not sure if it's encoding or else now. But I would investigate soon and let you know :)

Reply:
Thanks!

Reply:
No worries at all. 

I just did a quick (not thorough) investigation and found there is issue with the unit test, and wondering why this didn't show up in Unix/Windows runners at all...

The buggy test is the following (I removed unrelated part):
```python
class TestLobsterin(TestCase):
    def test_dict_functionality(self):
        Lobsterinfromfile = Lobsterin.from_file(f"{TEST_FILES_DIR}/cohp/lobsterin.1")
        assert Lobsterinfromfile.get("COHPstartEnergy") == -15.0
```

But the `from_file` method (wondering why this looks so complex) actually `lower()` all keys:
```python
    @classmethod
    def from_file(cls, lobsterin: str) -> Self:
        """
        Args:
            lobsterin (str): path to lobsterin.

        Returns:
            Lobsterin object
        """
        with open(lobsterin, mode="rt") as file:
            data = file.read().split("\n")
        if len(data) == 0:
            raise OSError("lobsterin file contains no data.")
        Lobsterindict: dict[str, Any] = {}

        for datum in data:
            # Remove all comments
            if not datum.startswith(("!", "#", "//")):
                pattern = r"\b[^!#//]+"  # exclude comments after commands
                if matched_pattern := re.findall(pattern, datum):
                    raw_datum = matched_pattern[0].replace("\t", " ")  # handle tab in between and end of command
                    key_word = raw_datum.strip().split(" ")  # extract keyword
                    if len(key_word) > 1:
                        # check which type of keyword this is, handle accordingly
                        if key_word[0].lower() not in [datum2.lower() for datum2 in Lobsterin.LISTKEYWORDS]:
                            if key_word[0].lower() not in [datum2.lower() for datum2 in Lobsterin.FLOAT_KEYWORDS]:
                                if key_word[0].lower() not in Lobsterindict:
                                    Lobsterindict[key_word[0].lower()] = " ".join(key_word[1:])
                                else:
                                    raise ValueError(f"Same keyword {key_word[0].lower()} twice!")
                            elif key_word[0].lower() not in Lobsterindict:
                                Lobsterindict[key_word[0].lower()] = float(key_word[1])
                            else:
                                raise ValueError(f"Same keyword {key_word[0].lower()} twice!")
                        elif key_word[0].lower() not in Lobsterindict:
                            Lobsterindict[key_word[0].lower()] = [" ".join(key_word[1:])]
                        else:
                            Lobsterindict[key_word[0].lower()].append(" ".join(key_word[1:]))
                    elif len(key_word) > 0:
                        Lobsterindict[key_word[0].lower()] = True

        return cls(Lobsterindict)
```

I'm sure you should have noticed the problem now, in unit test it stills get the value with the original key: `assert Lobsterinfromfile.get("COHPstartEnergy") == -15.0`, and it's thus default to `None`. I'm now just wondering why this doesn't show up elsewhere?

Reply:
I rewrote the get part, right? It should use the lower version of it? I am not sure I can follow.

Reply:
Probably one can simplify quite a lot here as well....

Reply:
>  It should use the lower version of it? 

Yes the unit test should use the lowered key `Lobsterinfromfile.get("cohpstartenergy")`, otherwise it would get the default `None`. 

> Probably one can simplify quite a lot here as well....

I would be more than happy to help with this but not in this PR. I'm not intending to include any functional changes (perhaps including code simplication) here, which might make the whole PR hard to track and review 😄

Reply:
> This is actually a test to test this?

Yes, see here: https://github.com/materialsproject/pymatgen/blob/37d1066a8f45b24dff8cf8c18b2549648897afce/tests/io/lobster/test_inputs.py#L1772-L1787

Reply:
No, it should not use the lower version of it. This is wrong.

Reply:
My guess is that this part here is not working on your machine:
https://github.com/DanielYang59/pymatgen/blob/9ec81411c35b86fbd3b352bd2232139eb30a29db/pymatgen/io/lobster/inputs.py#L151

I am not sure why.

Reply:
In any case, I have to go through this implementation to make sure everything is consistent. I feel a few things aren't. Nevertheless, this unit test should work.

Reply:
I'm still using the code from the master branch, I'm not sure if you have made any correction or not.

I minimized the unit test:
```python
class TestLobsterin(TestCase):
    def test_dict_functionality(self):
        Lobsterinfromfile = Lobsterin.from_file(f"{TEST_FILES_DIR}/cohp/lobsterin.1")
        print(f"{Lobsterinfromfile=}")
        print(Lobsterinfromfile.get("COHPstartEnergy"))
        assert Lobsterinfromfile.get("COHPstartEnergy") == -15.0
```

And it returns:
```python
=========================================================== FAILURES ============================================================
_____________________________________________ TestLobsterin.test_dict_functionality _____________________________________________

self = <lobster.test_inputs.TestLobsterin testMethod=test_dict_functionality>

    def test_dict_functionality(self):
        Lobsterinfromfile = Lobsterin.from_file(f"{TEST_FILES_DIR}/cohp/lobsterin.1")
        print(f"{Lobsterinfromfile=}")
        print(Lobsterinfromfile.get("COHPstartEnergy"))
>       assert Lobsterinfromfile.get("COHPstartEnergy") == -15.0
E       AssertionError

tests/io/lobster/test_inputs.py:18: AssertionError
----------------------------------------------------- Captured stdout call ------------------------------------------------------
Lobsterinfromfile={'cohpstartenergy': -15.0, 'cohpendenergy': 5.0, 'basisset': 'pbeVaspFit2015', 'gaussiansmearingwidth': 0.1, 'basisfunctions': ['Fe 3d 4p 4s', 'Co 3d 4p 4s'], 'skipdos': True, 'skipcohp': True, 'skipcoop': True, 'skippopulationanalysis': True, 'skipgrosspopulation': True}
None
===================================================== slowest 30 durations ======================================================

(3 durations < 0.005s hidden.  Use -vv to show these durations.)
1 failed in 1.64s
```

Reply:
Yeah, but only on your machine.

Reply:
I did not make any modifications. The whole unit test passes for me.

Reply:
Yes it's weird and I don't know why. I saw you have `__get/setitem__` methods to handle case sensitivity but seem not working on my side. Maybe @janosh would have some insights?

In any case, we might enjoy the rest of the Sunday and come back tomorrow with clearer eyes. Thanks for trying to fix this!

Reply:
Yes 😅. I will just make the del consistent. There is an issue with the logic there and then we check next week 

Thanks!

Reply:
Thanks!

I would try to give a hand on simplifying the logic here when we finish cleaning up unbound vars on this one.

Enjoy the weekend!